File: java/poker/src/main/java/com/duchen/poker/model/type/BombTypeChecker.java
package com.duchen.poker.model.type;

import java.util.Arrays;
import java.util.List;

public class BombTypeChecker implements TypeChecker {

    @Override
    public CardGroupType getType() {
        return CardGroupType.BOMB_CARD;
    }

    @Override
    public List<Integer> getMyPossibleCardCount() {
        return Arrays.asList(4);
    }

    @Override
    public int getMaxCardIfIsMyType(List<Integer> cards) {
        int card = cards.get(0);
        for (int data : cards) {
            if (data != card) {
                return -1;
            }
        }
        return card;
    }
}

File: java/poker/src/main/kotlin/poker/model/type/BombTypeChecker.kt
package poker.model.type

import java.util.Arrays

class BombTypeChecker : TypeChecker {

    override val type: CardGroupType
        get() = CardGroupType.BOMB_CARD

    override val myPossibleCardCount: List<Int>
        get() = Arrays.asList(4)

    override fun getMaxCardIfIsMyType(cards: MutableList<Int>): Int {
        val card = cards[0]
        return if (cards.any { it != card }) -1 else card
    }
}
